### ISSUE TITLE
Inconsistency between Linux and macOS/Windows clients for REALs

### ISSUE THREAD

**Issue by jasperswallen** [eerimoq/asn1tools#134]
I've been having some _very_ strange behavior that I seem to have tracked down to float inconsistency handling. I'm not entirely sure which implementation is correct, since I'm not an expert at reading OER ASN.1. To my untrained eye, the macOS version seems too short (6 hex digits = 24 bits), while the Linux version seems too long (12 hex digits = 48 bits). I expect the spec I wrote to hold exactly 32 bits (8 hex digits).

Minimal reproducible code:

main.py:
```python
import asn1tools


def main():
    spec = asn1tools.compile_files(['Spec.asn'], codec='oer')

    result = spec.encode('Example', data={'z': 0.1}, check_constraints=True)
    print(result)
    print(f"Decoded: {result.hex()}")

    decoded = spec.decode('Example', result, True)
    print(decoded)

main()
```

Spec.asn:
```ASN.1
Spec DEFINITIONS AUTOMATIC TAGS ::=
BEGIN

Float ::=
  REAL
    (WITH COMPONENTS {
       mantissa  (-16777215..16777215),
       base      (2),
       exponent  (-149..104)
     })

Example ::= SEQUENCE
{
    z Float
}

END
```

On macOS and Windows, I get this as the output:
```
➜ python3 main.py
bytearray(b'=\xcc\xcc\xcd')
Decoded: 3dcccccd
{'z': 0.10000000149011612}
```

On Linux, I get this as the output:
```
➜ python3 main.py
bytearray(b'\t\x80\xc9\x0c\xcc\xcc\xcc\xcc\xcc\xcd')
Decoded: 0980c90ccccccccccccd
{'z': 0.1}
```

Tested with the following environments:
* macOS 12.1 beta, Python 3.9.9 (Homebrew), asn1tools 0.159.0 installed from wheel
* Kubuntu 21.04, Python 3.9.5 (apt), asn1tools 0.159.0 installed from wheel
* Windows Server 2019 Essentials, Python 3.9.6, asn1tools 0.159.0 (also tested with 0.158.0)

**Comment by eerimoq** [comment]
Which version of pyparsing do you have on those systems? asn1tools uses pyparsing to parse asn1-files. My guess is that asn1tools only works with pyparsing version 2.x.y, not 3.m.n.

**Comment by jasperswallen** [comment]
Yup, on Kubuntu I had 3.0.6 installed. I installed 2.4.7 instead, and now it properly sizes floats!

**Comment by eerimoq** [comment]
The latest version, 0.160.0, of asn1tools works with both old and new pyparsing.
